internal/logger/grpclogger.go

```diff
@@ -7,71 +7,92 @@ SPDX-License-Identifier: AGPL-3.0-only
 package logger
 
 import (
+	"context"
 	"fmt"
+	"log/slog"
+	"os"
+	"runtime"
+	"time"
 
-	"go.uber.org/zap"
 	"google.golang.org/grpc/grpclog"
 )
 
-func replaceGRPCLogger(log *zap.Logger) {
+func replaceGRPCLogger(log *slog.Logger) {
 	gl := &grpcLogger{
-		logger:    log.With(zap.String("system", "grpc"), zap.Bool("grpc_log", true)).WithOptions(zap.AddCallerSkip(2)),
+		logger:    log.With(slog.String("system", "grpc"), slog.Bool("grpc_log", true)),
 		verbosity: 0,
 	}
 	grpclog.SetLoggerV2(gl)
 }
 
+func (l *grpcLogger) log(level slog.Level, args ...interface{}) {
+	var pcs [1]uintptr
+	runtime.Callers(3, pcs[:])
+	r := slog.NewRecord(time.Now(), level, fmt.Sprint(args...), pcs[0])
+	_ = l.logger.Handler().Handle(context.Background(), r)
+}
+
+func (l *grpcLogger) logf(level slog.Level, format string, args ...interface{}) {
+	var pcs [1]uintptr
+	runtime.Callers(3, pcs[:])
```

Comment:
```suggestion
	runtime.Callers(7, pcs[:])
	r := slog.NewRecord(time.Now(), level, fmt.Sprint(args...), pcs[0])
	_ = l.logger.Handler().Handle(context.Background(), r)
}

func (l *grpcLogger) logf(level slog.Level, format string, args ...interface{}) {
	var pcs [1]uintptr
	runtime.Callers(7, pcs[:])
```
Caller skip of 7 was needed to get the actual line in the grpc package that was making the logging call.
I'm however unsure if we want to bother logging the exact line for grpc, since the caller depth may change on new releases of the grpc library.
Caller skip of 3 would make sure we skip just our own functions and log the location of the top level grpc code.

Reply:
Is there a preferred option? I chose three since it should be more consistent with the code before the conversion to slog, but I can see why it could be beneficial to log the actual line instead of the top level code.